File: Backend/src/main/java/com/blazarczyk/praca/model/request/CreateProjectRequest.java
package com.blazarczyk.praca.model.request;

import com.blazarczyk.praca.model.databse.Link;
import com.blazarczyk.praca.model.databse.ProjectDetail;
import com.blazarczyk.praca.model.databse.Tag;
import com.blazarczyk.praca.model.json.LinkJson;
import com.blazarczyk.praca.model.json.ProjectDetailJson;
import com.blazarczyk.praca.model.json.TagJson;

import javax.persistence.*;
import java.util.Set;

public class CreateProjectRequest {

    private String name;
    private String description;
    private String imageURL;
    private boolean publicProject;
    private Set<ProjectDetailJson> projectDetails;
    private Set<LinkJson> links;
    private Set<TagJson> tags;

}

File: Frontend/src/app/model/request/CreateProjectRequest.ts
import {ProjectDetail} from '../ProjectDetail';
import {Link} from '../Link';
import {Tag} from '../Tag';

export class CreateProjectRequest {
  constructor(public name: string, public description: string, public imageURL: string,
              public publicProject: boolean, public projectDetails: ProjectDetail[],
              public links: Link[], public tags: Tag[]) {}
}
